src/clientSideScene/ClientSideSceneComp.tsx

```diff
@@ -96,15 +96,29 @@ export const ClientSideScene = ({
     canvas.appendChild(sceneInfra.renderer.domElement)
     canvas.appendChild(sceneInfra.labelRenderer.domElement)
     sceneInfra.animate()
-    canvas.addEventListener('mousemove', sceneInfra.onMouseMove, false)
+    canvas.addEventListener(
+      'mousemove',
+      toSync(sceneInfra.onMouseMove, reportRejection),
+      false
+    )
     canvas.addEventListener('mousedown', sceneInfra.onMouseDown, false)
-    canvas.addEventListener('mouseup', sceneInfra.onMouseUp, false)
+    canvas.addEventListener(
+      'mouseup',
+      toSync(sceneInfra.onMouseUp, reportRejection),
```

Comment:
@jtran re https://github.com/KittyCAD/modeling-app/pull/3860#discussion_r1768630322

I think this is the appropriate way to make us `toSync` right?
I had to change the return types of these callbacks to be promisy/async, but these went listeners need to be synchronous.

Reply:
Yes, awesome! I didn't expect you to fix all the old ignores too, but this is great. Ideally, we'd get rid of them all eventually.